app/uk/gov/hmrc/merchandiseinbaggagefrontend/model/core/Declaration.scala

```diff
@@ -189,6 +190,7 @@ case class DeclarationJourney(sessionId: SessionId,
   val declarationIfRequiredAndComplete: Option[Declaration] = {
     val discardedAnswersAreCompleteAndRequireADeclaration =
       maybeGoodsDestination.isDefined &&
+        maybeImportOrExportGoodsFromTheEUViaNorthernIreland.contains(No) &&
```

Comment:
I think this is still wrong. maybeImportOrExportGoodsFromTheEUViaNorthernIreland only needs two be 'No' if maybeGoodsDestination == Some(NorthernIreland)

And so I think the tests you added below aren't quite right also